org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/testexecution/TestSuiteExecutorIntegrationTest.xtend

```diff
@@ -384,6 +384,71 @@ class TestSuiteExecutorIntegrationTest extends AbstractPersistenceIntegrationTes
 		
 	}
 	
+	@Test
+	def void testThatSubStepScreenshotDetailsAreProvided() {
+		val testKey = TestExecutionKey.valueOf('0-0')
+		val childKeys = #['IDXY', 'IDXZ', 'IDYZ']
+		val testFile = 'test.tcl'
+		val userDir = workspaceRoot.newFolder(userId)
+		workspaceRoot.newFile(userId + '/' + testFile)
+		workspaceRoot.newFolder(userId, 'logs')
+		workspaceRoot.newFile(userId + '''/logs/testrun.«testKey».299900000000.yaml''') => [
+			executable = true
+			JGitTestUtil.write(it, '''
+				"testRuns":
+				- "source": "test.tcl"
+				  "testRunId": "1"
+				  "children" :
+				  - "node": "TEST"
+				    "id": "ID1"
+				    "children":
+				    - "node": "SPECIFICATION"
+				      "id": "ID2"
+				      "message": "hello"
+				      "children":
+				      - "id": "IDXY"
+				      - "id": "IDXZ"
+				      - "id": "IDYZ"
+				    - "node": "SPECIFICATION"
+				      "id": "ID3"
+			''')
+		]
+		workspaceRoot.newFile(userId + '/gradlew') => [
+			executable = true
+			JGitTestUtil.write(it, '''
+				#!/bin/sh
+				echo "Running mock gradlew script from working directory $(pwd)"
+				set -x
+				targetDir=".testexecution/artifacts/«testKey.suiteId»/«testKey.suiteRunId»/1"
+				mkdir -p ${targetDir}
+				«FOR id: childKeys»
+				printf '"screenshot": "screenshots/test/«id».png"\n' > ${targetDir}/«id».yaml
+				«ENDFOR»
+			''')
+		]
+
+		val response = createLaunchNewRequest().post(Entity.entity(#[testFile], MediaType.APPLICATION_JSON_TYPE))
+		response.status.assertEquals(Status.CREATED.statusCode)
+		createTestStatusRequest(testKey).get // wait for completion
+		childKeys.forall[new File(userDir, '''.testexecution/artifacts/0/0/1/«it».yaml''').exists]
+			.assertTrue('Mocked process did not write yaml file with screenshot information.')
+		
+		// when
+		val result = createNodeRequest(testKey.deriveWithCaseRunId("1").deriveWithCallTreeId('ID2')).get.readEntity(String)
+
+		// then
+		val propertiesList = newArrayList
```

Comment:
why not directly assign the list? newArrayList and addAll is not really necessary, is it?

Reply:
propertiesList wouldn't be a list then, but an iterable (which doesn't happen to be a list at runtime, so I cannot cast it, either). Those don't have a get method. Xtend will falsely try to match the get methods from ArrayExtensions, but that also leaves an ambiguity the compiler will complain about, because the generic type cannot be narrowed down sufficiently to make an unambiguous match.

So, long story short, wrapping in a list was the simplest workaround to express what I wanted (at least in the context of a test).